### PR TITLE
Add Java sample application

### PR DESCRIPTION
**Pull request by tifflhl**
- Add .gitignore
- Add Java sample client application and README.md in java folder
- Add Java test and build to a new Github workflow

### CODE REVIEW

File: java/README.md [hunk 1]
```diff
@@ -1,27 +1,60 @@
 # Java Arrow Flight Client Application Example
 ![Build Status](https://github.com/dremio-hub/arrow-flight-client-examples/workflows/java-build/badge.svg)
 
+This lightweight Java client application connects to the Dremio Arrow Flight server endpoint. It requires the username and password for authentication. Developers can use admin or regular user credentials for authentication. Any datasets in Dremio that are accessible by the provided Dremio user can be queried. By default, the hostname is `localhost` and the port is `32010`. Developers can change these default settings by providing the hostname and port as arguments when running the client. Moreover, the tls option can be provided to establish an encrypted connection. 
+> Note that: Trusted certificates must be provided when the tls option is enabled.
```
Comment (csolonynka) [inline java/README.md:5]: I would remove "that" and just have Note: Trusted certificates must be provided...

File: java/README.md [hunk 2]
```diff
@@ -1,27 +1,60 @@
...
+
 ### Pre-requisites
```
Comment (csolonynka) [inline java/README.md:7]: Prerequisites is a word, you don't have to add the dash.

File: java/README.md (hunk 1 above)
Comment (tifflhl) [reply java/README.md:5]: Addressed.

File: java/README.md (hunk 2 above)
Comment (tifflhl) [reply java/README.md:7]: Will be addressed in new commit.

**Review by csolonynka — APPROVED** [review]

**Review by csolonynka — APPROVED** [review]

**Review by csolonynka — APPROVED** [review]
Looks Great!

File: java/README.md [hunk 3]
```diff
@@ -0,0 +1,60 @@
+# Java Arrow Flight Client Application Example
+![Build Status](https://github.com/dremio-hub/arrow-flight-client-examples/workflows/java-build/badge.svg)
+
+This lightweight Java client application connects to the Dremio Arrow Flight server endpoint. It requires the username and password for authentication. Developers can use admin or regular user credentials for authentication. Any datasets in Dremio that are accessible by the provided Dremio user can be queried. By default, the hostname is `localhost` and the port is `32010`. Developers can change these default settings by providing the hostname and port as arguments when running the client. Moreover, the tls option can be provided to establish an encrypted connection. 
+> Note that: Trusted certificates must be provided when the tls option is enabled.
+
+### Prerequisites
+- Java 8
```
Comment (kylep-dremio) [inline java/README.md:8]: Java 8? or Java 8 or above?

File: java/src/main/java/com/adhoc/flight/QueryRunner.java [hunk 4]
```diff
@@ -0,0 +1,254 @@
+package com.adhoc.flight;
```
Comment (kylep-dremio) [inline java/src/main/java/com/adhoc/flight/QueryRunner.java:1]: Should there be a license header?

File: java/src/main/java/com/adhoc/flight/QueryRunner.java [hunk 5]
```diff
@@ -0,0 +1,254 @@
...
+
+        @Parameter(names = {"-h", "--help"},
+                description = "show usage", help=true)
+        public boolean help = false;
+    }
+
+    /**
+     * Runs a self contained demo to authenticate and query a Dremio Flight Server Endpoint.
+     *
+     * @throws Exception
```
Comment (kylep-dremio) [inline java/src/main/java/com/adhoc/flight/QueryRunner.java:79]: We should have a description of why this is thrown.

File: java/src/main/java/com/adhoc/flight/QueryRunner.java [hunk 6]
```diff
@@ -0,0 +1,254 @@
...
+             * Drop Demo Table
+             */
+            System.out.println("[INFO] [STEP 5]: Drop demo table.");
+            client.runQuery(DROP_DEMO_TABLE, schemaCallOption);
+            System.out.println("[INFO] Dropped $scratch.dremio_flight_demo_table successfully");
+        } catch (Exception ex) {
+            System.out.println("[ERROR] Exception: " + ex.getMessage());
+            ex.printStackTrace();
+        } finally {
+            AutoCloseables.close(client);
```
Comment (kylep-dremio) [inline java/src/main/java/com/adhoc/flight/QueryRunner.java:145]: Seems like we should use try-with-resources instead.

File: java/README.md (hunk 3 above)
Comment (tifflhl) [reply java/README.md:8]: Just Java 8, to match Dremio's official Java support.

File: java/src/main/java/com/adhoc/flight/QueryRunner.java (hunk 6 above)
Comment (tifflhl) [reply java/src/main/java/com/adhoc/flight/QueryRunner.java]: Addressed

File: java/src/main/java/com/adhoc/flight/QueryRunner.java (hunk 5 above)
Comment (tifflhl) [reply java/src/main/java/com/adhoc/flight/QueryRunner.java]: Addressed.

File: java/src/main/java/com/adhoc/flight/QueryRunner.java (hunk 4 above)
Comment (tifflhl) [reply java/src/main/java/com/adhoc/flight/QueryRunner.java:16]: Addressed.

File: .github/workflows/python-workflow.yaml [hunk 7]
```diff
@@ -67,14 +67,17 @@ jobs:
           flake8 . --count --select=E9,F63,F7,F82 --show-source --statistics
           # Exit-zero treats all errors as warnings. The GitHub editor is 127 chars wide
           flake8 . --count --exit-zero --max-complexity=10 --max-line-length=127 --statistics
+      # Below test steps are temporarily tunred off until Python sample client application is
```
[the remark this answers is not in the document]
Comment (tifflhl) [reply .github/workflows/python-workflow.yaml]: Addressed.

File: java/src/main/java/com/adhoc/flight/client/AdhocFlightClient.java [hunk 8]
```diff
@@ -0,0 +1,215 @@
...
+                for (int rowIndex = 0; rowIndex < rowCount; rowIndex++) {
+                    if (values.size() - 1 < rowIndex) {
+                        values.add(new Object[columnCount]);
+                    }
+                    final Object[] rowValues = values.get(rowIndex);
+                    final Object value = fieldVector.getObject(rowIndex);
+                    rowValues[columnIndex] = value;
+                }
+            }
+            AutoCloseables.close(root);
```
Comment (kylep-dremio) [inline java/src/main/java/com/adhoc/flight/client/AdhocFlightClient.java:202]: Seems like these should use try-with-resources as well?

**Review by kylep-dremio — APPROVED** [review]

**Pull request merged**